radicle/src/identity/project.rs

```diff
@@ -20,42 +25,189 @@ pub enum ProjectError {
 }
 
 /// A "project" payload in an identity document.
-#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
 #[serde(rename_all = "camelCase")]
 pub struct Project {
     /// Project name.
-    pub name: String,
+    name: String,
     /// Project description.
-    pub description: String,
+    description: String,
     /// Project default branch.
-    pub default_branch: BranchName,
+    default_branch: BranchName,
+}
+
+impl<'de> Deserialize<'de> for Project {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        #[derive(Deserialize)]
+        #[serde(field_identifier, rename_all = "camelCase")]
+        enum Field {
+            Name,
+            Description,
+            DefaultBranch,
+        }
+
+        struct ProjectVisitor;
+
+        impl<'de> Visitor<'de> for ProjectVisitor {
+            type Value = Project;
+
+            fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+                formatter.write_str("Project payload")
```

Comment:
Should we use the type id here for the string to be more specific? ie. `xyz.radicle.project`

Reply:
Ya, that makes sense!